game-core/src/main/java/games/strategy/triplea/delegate/MoveDelegate.java

```diff
@@ -189,13 +189,11 @@ public void end() {
       removeAirThatCantLand();
     }
 
-    // WW2V2/WW2V3, fires at end of combat move
-    // WW2V1, fires at end of non combat move
-    if (GameStepPropertiesHelper.isFireRockets(data)) {
-      if (needToDoRockets && TechTracker.hasRocket(bridge.getPlayerId())) {
-        RocketsFireHelper.fireRockets(bridge, bridge.getPlayerId());
-        needToDoRockets = false;
-      }
+    // WW2V2/WW2V3, fires at end of combat move, for legacy maps only
+    // WW2V1, fires at end of non combat move, for legacy maps only
+    if (needToDoRockets && GameStepPropertiesHelper.isNonCombatMove(data, true)) {
```

Comment:
Why are we checking for isNonCombatMove here?

Reply:
The comment doesn't help? Perhaps the bit about V2/V3 is distracting?

V1 rockets fire at the end of NCM.